File: apps/android/feature/workouts/domain/src/main/java/com/scribblefit/feature/workouts/domain/WorkoutRepository.kt
package com.scribblefit.feature.workouts.domain

import com.scribblefit.core.model.Workout
import kotlinx.coroutines.flow.Flow

interface WorkoutRepository {
    suspend fun saveWorkout(workout: Workout): Long
    fun getWorkoutByDate(date: Long): Flow<Workout?>
    fun getWorkoutById(workoutId: Long): Flow<Workout?>
    fun getWorkoutsInRange(startDate: Long, endDate: Long): Flow<List<Workout>>
}

File: apps/ios/ScribbleFit/Sources/Core/Model/WorkoutRepository.swift
import Foundation

@MainActor
public protocol WorkoutRepository: Sendable {
    func getWorkout(id: UUID) async throws -> Workout?
    func saveWorkout(_ workout: Workout) async throws
    func deleteWorkout(id: UUID) async throws
    func getWorkouts(for date: Date) -> AsyncStream<[Workout]>
    func getWorkoutsInRange(startDate: Date, endDate: Date) async throws -> [Workout]
}
